Does not recognize additional --builtins

[As per the docs](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-builtins), it is possible to add builtins to flake8 by passing the `--builtins` flag or through config files.

However, these are not recognized by flake8-builtins:

```python
import gettext

gettext.install("minimal_repro")


def id(s):
    return s


def _(s):
    return s


print(id("foo"))
print(_("bar"))
```

```bash
$ python ./no_custom_builtins.py
foo
bar
$ flake8 ./no_custom_builtins.py --builtins=_
./no_custom_builtins.py:1:1: D100 Missing docstring in public module
./no_custom_builtins.py:6:1: A001 variable "id" is shadowing a python builtin
./no_custom_builtins.py:6:1: D103 Missing docstring in public function
```

A001 should be reported for both functions.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -6,8 +6,8 @@ Changelog
 2.0.2 (unreleased)
 ------------------
 
-- Nothing changed yet.
-
+- Honor `--builtins` option from flake8 #73.
+  [gforcada]
 
 2.0.1 (2022-11-01)
 ------------------

--- a/flake8_builtins.py
+++ b/flake8_builtins.py
@@ -48,6 +48,9 @@ class BuiltinsChecker:
         cls.names = {
             a[0] for a in inspect.getmembers(builtins) if a[0] not in cls.ignore_list
         }
+        flake8_builtins = getattr(options, 'builtins', None)
+        if flake8_builtins:
+            cls.names.update(flake8_builtins)
 
     def run(self):
         tree = self.tree

--- a/run_tests.py
+++ b/run_tests.py
@@ -10,13 +10,16 @@ from flake8_builtins import BuiltinsChecker
 
 class FakeOptions:
     builtins_ignorelist = []
+    builtins = None
 
-    def __init__(self, ignore_list=''):
+    def __init__(self, ignore_list='', builtins=None):
         if ignore_list:
             self.builtins_ignorelist = ignore_list
+        if builtins:
+            self.builtins = builtins
 
 
-def check_code(source, expected_codes=None, ignore_list=None):
+def check_code(source, expected_codes=None, ignore_list=None, builtins=None):
     """Check if the given source code generates the given flake8 errors
 
     If `expected_codes` is a string is converted to a list,
@@ -24,6 +27,9 @@ def check_code(source, expected_codes=None, ignore_list=None):
 
     If `ignore_list` is provided, it should be a list of names
     that will be ignored if found, as if they were a builtin.
+
+    If `builtins` is provided, it should be a list of names
+    that will be reported if found, as if they were a builtin.
     """
     if isinstance(expected_codes, str):
         expected_codes = [expected_codes]
@@ -33,7 +39,7 @@ def check_code(source, expected_codes=None, ignore_list=None):
         ignore_list = []
     tree = ast.parse(textwrap.dedent(source))
     checker = BuiltinsChecker(tree, '/home/script.py')
-    checker.parse_options(FakeOptions(ignore_list=ignore_list))
+    checker.parse_options(FakeOptions(ignore_list=ignore_list, builtins=builtins))
     return_statements = list(checker.run())
 
     assert len(return_statements) == len(expected_codes)
@@ -210,6 +216,11 @@ def test_default_ignored_names():
 def test_custom_ignored_names():
     source = 'copyright = 4'
     check_code(source, ignore_list=('copyright',))
+
+
+def test_flake8_builtins():
+    source = 'consider_this_builtin = 4'
+    check_code(source, ['A001'], builtins=('consider_this_builtin',))
 
 
 def test_for_loop_variable():

